### DISCUSSION TITLE
How to login a customer?

### DISCUSSION THREAD

**Discussion by hallleron** [pimcore/pimcore#9520] [General]
Hey,

I created a login form where the customer has to enter username and password. This data is then posted to my controller where the following action is executed:
```

public function authenticateAction(Request $request): Response
{
        $req = $request->request;
        $username = $req->get('username');
        $password = $req->get('password');

        $customer = $this->_customer->getActiveCustomerByEmail($username);
        if ($customer === false) {
            return new Response('Customer not found');
        }
}
```

Btw `$this->_customer` is an instance of `CustomerManagementFrameworkBundle\CustomerProvider\CustomerProviderInterface`.

But now I am stuck. How can I login the $customer when it isn't false? I searched on Google for quite a while without any solution.

Thanks in advance!

**Comment by jdreesen** [comment]
Did you read https://symfony.com/doc/current/security.html? Not sure if the new authentication manager already works with Pimcore, so maybe you have to use the "old" guards method.

**Reply by hallleron** [reply]
I am confused, isn't "User" reffered to a backend user only? I want to login with a customer. And what exactly is that guards method if I may ask? Thank you in advance!
